### DISCUSSION TITLE
caddyfile

### DISCUSSION THREAD

**Discussion by LuisGB2212** [dunglas/mercure#664] [General]
Please help

When executing docker run with my caddyfile file, I get an error.

![caddy_error](https://user-images.githubusercontent.com/47278714/179184053-57775361-5c93-4f56-bb24-609002c376d0.png)

My file Caddyfile

{
	{$GLOBAL_OPTIONS}
}

{$SERVER_NAME:localhost}

log

route {
	redir / /.well-known/mercure/ui/
	encode zstd gzip

	mercure {
		# Transport to use (default to Bolt)
		transport_url {$MERCURE_TRANSPORT_URL:bolt:///data/mercure.db}
		# Publisher JWT key
		publisher_jwt {env.MERCURE_PUBLISHER_JWT_KEY} {env.MERCURE_PUBLISHER_JWT_ALG}
		# Subscriber JWT key
		subscriber_jwt {env.MERCURE_SUBSCRIBER_JWT_KEY} {env.MERCURE_SUBSCRIBER_JWT_ALG}
		# Permissive configuration for the development environment
		cors_origins *
		publish_origins *
		demo
		anonymous
		subscriptions
		# Extra directives
		{$MERCURE_EXTRA_DIRECTIVES}
	}

	respond /healthz 200

	respond "Not Found" 404
}

**Comment by EvilSpartans** [comment]
You need to add order mercure after encode like here : 

```
{
	order mercure after encode
	{$GLOBAL_OPTIONS}
}
```
